src/content/tutorials/en/intro-to-p5-strands.mdx

```diff
@@ -0,0 +1,744 @@
+}
+
+function draw() {
+	background(0);
+	orbitControl();
+	shader(instancingShader)
+  fill(255, 100, 150)
+  strokeShader(instancingStrokeShader)
+	model(particleModel, 1000);
+}
+`} />
+
+By adding `time / 10000` to the phi angle and modifying the radius with `sin()`, we create animated particles that move around a pulsing, stretched sphere. Play around and see what other shapes you can create.
+
+<Callout title='Experiment!'>
+Try replacing `sin()` with `tan()`, `acosh()`, or combinations of functions. Small changes in shader code often create dramatically different visual effects, so experimentation goes a long way.
+</ Callout>
+
+### Fresnel effect
+If you've ever seen a material in a 3D render which appears to glow at the edges, or noticed how the light reflections appear to change on virtual water as you move your viewpoint, you were seeing the Fresnel effect. This effect changes how materials look when viewed at an angle.
+
+The Fresnel effect will be checking which parts of the shape are pointing away from the camera. For this reason, it is helpful for us to work in camera space. In camera space, also known as view space:
+- The camera is position at the origin `(0, 0, 0)`
+- The camera looks along the negative Z-axis
+- All 3D positions are relative to the camera's perspective
+
+<Callout title='Camera Space'>
+This is another relative view of the virtual world, where the camera is positioned at `(0, 0)`, so everything else is position relative to it.
+</ Callout>
+
+This perspective makes it easier to determine how surfaces appear to the viewer:
+
+```js
+function fresnelCallback() {
+	getCameraInputs((inputs) => {
+		// The normal vector on the sphere's surface
+		let normalVector = normalize(inputs.normal);
+		// This creates a vector pointing from the surface to the camera
+		let viewVector = normalize(-inputs.position);
+		// ...
+		return inputs;
+	})
+}
+```
+
+The line let viewVector = normalize(-inputs.position) might seem counterintuitive, so let's break it down:
+
+1. In camera space, the camera is at `(0, 0, 0)`
+2. inputs.position gives us the position of the current vertex being processed
+3. When we negate this (`-inputs.position`), we get a vector pointing from the vertex toward the camera
+4. `normalize()` converts this to a unit vector (where every component is between 0 - 1), making it useful for direction calculations regardless of distance
+
+#### Calculating the Fresnel factor
+
+The core of the Fresnel effect is comparing two directions:
+1. The surface normal (where the surface is "pointing")
+2. The view direction (where the camera is relative to the surface)
+
+```js
+let base = 1 - dot(normalVector, viewVector);
+let fresnel = pow(base, 2);
+```
+The dot product between these directions measures how 'edge-like' each point on our sphere is. It returns the cosine of the angle between them. The values range from 1 (when they're parallel) to 0 when they're perpendicular, and -1 when they're opposite. 

```

Comment:
Might be worth reiterating what you said before about the dot product telling you how similar two vectors are, returning a value between -1 (opposite) and 1 (the same).